### ISSUE TITLE
GinBot spreading cards it doesn't have.

### ISSUE THREAD

**Issue by ichabod801** [ichabod801/t_games#369]
I am seeing this error a lot in tournament testing:

```
  File "C:\Users\Craig\Documents\t_games\card_games\gin_rummy_game.py", line 1124, in player_action
    go = self.handle_cmd(move)
  File "C:\Users\Craig\Documents\t_games\other_cmd.py", line 174, in handle_cmd
    return getattr(self, method)(arguments.strip())
  File "C:\Users\Craig\Documents\t_games\card_games\gin_rummy_game.py", line 817, in do_knock
    attack_melds, attack_deadwood = self.spread(attacker)
  File "C:\Users\Craig\Documents\t_games\card_games\gin_rummy_game.py", line 1293, in spread
    player.error('You do not have all of those cards.')
  File "C:\Users\Craig\Documents\t_games\player.py", line 553, in error
    raise BotError(text.strip())
t_games.player.BotError: You do not have all of those cards.
```

**Comment by ichabod801** [comment]
He appears to be cross tracking things. One error shows him having JC, 9D, 8S, 6D, JS, 8H, TD, 8D, 7D, JD, and trying to do a last meld of 9D, TD, JD. But note that he also had (8S, 8H, 8D), and (JC JS JD), and all sorts of ways to do the run. If he had cross tracked the jacks with the straight, that could have caused the problem.

**Comment by ichabod801** [comment]
I think the run is the problem. He's got a five card run he isn't connecting up and is tracking it as two three card runs.

**Comment by ichabod801** [comment]
Found the problem, if not the fix. If you have a run that gets broken twice by two sets, the second break is saved not taking into account the first break. Then when it resets the tracking, one card can be in multiple places in the tracking.

**Comment by ichabod801** [comment]
I could refuse to break a run twice. But that could cause problems. Say you had 2D 2H 2S 3S 4S 4C 4D. If you break at the twos, then you are missing the set of fours. Maybe don't break a run twice if there's a remainder?

**Comment by ichabod801** [comment]
Put the previous comment example in as a test case to make sure any fix doesn't break it.

**Comment by ichabod801** [comment]
I think this is fixed by 977aba9, but I'm holding it until I can run a tournament with no errors.

**Comment by ichabod801** [comment]
GinBot is running tournaments all the way through.
